src/wp-includes/media.php

```diff
@@ -1958,6 +1981,97 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 	return $image;
 }
 
+/**
+ * Adds optimization attributes to an `img` HTML tag.
+ *
+ * @since 6.3.0
+ *
+ * @param string $image   The HTML `img` tag where the attribute should be added.
+ * @param string $context Additional context to pass to the filters.
+ *
+ * @return string Converted `img` tag with optimization attributes added.
+ */
+function wp_img_tag_add_loading_optimization_attrs( $image, $context ) {
+	$image_elm = new WP_HTML_Tag_Processor( $image );
+	$image_elm->next_tag();
+	$width             = $image_elm->get_attribute( 'width' );
+	$height            = $image_elm->get_attribute( 'height' );
+	$loading_val       = $image_elm->get_attribute( 'loading' );
+	$fetchpriority_val = $image_elm->get_attribute( 'fetchpriority' );
+
+	// Bail early if both the priority attributes are already present.
+	if ( ! empty( $loading_val ) && ! empty( $fetchpriority_val ) ) {
+		return $image;
+	}
+
+	$optimization_attrs = wp_get_loading_optimization_attributes(
+		'img',
+		array(
+			'width'         => $width,
+			'height'        => $height,
+			'loading'       => $loading_val,
+			'fetchpriority' => $fetchpriority_val,
+		),
+		$context
+	);
+
+	// Retained for backward compatibility.
+	$loading_attrs_enabled = wp_lazy_loading_enabled( 'img', $context );
+
+	if ( empty( $loading_val ) && $loading_attrs_enabled ) {
+		/**
+		 * Filters the `loading` attribute value to add to an image. Default `lazy`.
+		 * This filter is added in for backward compatibility.
+		 *
+		 * Returning `false` or an empty string will not add the attribute.
+		 * Returning `true` will add the default value.
+		 * `true` and `false` usage supported for backward compatibility.
+		 *
+		 * @param string|bool $optimization_attrs['loading'] Current value for `loading` attribute for the image.
+		 * @param string      $image   The HTML `img` tag to be filtered.
+		 * @param string      $context Additional context about how the function was called or where the img tag is.
+		 */
+		$filtered_loading_attr = apply_filters(
+			'wp_img_tag_add_loading_attr',
+			isset( $optimization_attrs['loading'] ) ? $optimization_attrs['loading'] : false,
+			$image,
+			$context
+		);
+
+		/**
+		 * Validate the values after filtering.
+		 */
+		if ( isset( $optimization_attrs['loading'] ) && ! $filtered_loading_attr ) {
+			// unset loading attributes if $filtered_loading_attr is set to false.
+			unset( $optimization_attrs['loading'] );
+		} elseif ( in_array( $filtered_loading_attr, array( 'lazy', 'eager' ), true ) ) {
+			$optimization_attrs['loading'] = $filtered_loading_attr;
+		}
+
+		// fetchpriority='high' and loading='lazy' should be mutually exclusive.
+		if (
+		isset( $optimization_attrs['loading'] ) && isset( $optimization_attrs['fetchpriority'] ) &&
```

Comment:
```suggestion
		isset( $optimization_attrs['loading'], $optimization_attrs['fetchpriority'] ) &&
```
Nitpick: We can pass both to `isset()` which treats this as an `AND` gate.

Reply:
WordPress core still does not use this format. I don't remember the ticket number where I asked about the same.

Reply:
It does appear to be used in several places in Core (ThemeJSON, wpdb, community-events, post-template, update, user), and also in `media.php` which is touched by this PR [Ref](https://github.com/costdev/wordpress-develop/blob/530fd02fbe66789a121d641625f32a02e7211ac0/src/wp-admin/includes/media.php#L1660).

It's not a blocker by any means though, just seemed a bit cleaner than two separate conditions.